### PR TITLE
Adding new rule Do you have a clean git history?

### PR DESCRIPTION
**Pull request by GordonBeeming**
**Tip: Use [SSW Rule Writer GPT](https://chat.openai.com/g/g-cOvrRzEnU-ssw-rules-writer) for help with writing rules 🤖**
>
> 1. What triggered this change? (PBI link, Email Subject, conversation + reason, etc)

✏️ Creating a CTF on the topic

> 2. What was changed?

✏️ new rule

> 3. Did you do pair or mob programming (list names)?

✏️ partially with @danielmackay

### CODE REVIEW

File: rules/clean-git-history/rule.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+## Squashing Pull Requests
+
+`youtube: https://www.youtube.com/watch?v=0chZFIZLR_0`
+**Video: Git MERGE vs REBASE: Everything You Need to Know (4 min)**
+
+Squashing commits is the process of combining multiple commits into a single commit. This allows for the repository history to reflect changes to the application over time, rather than individual commits making up a feature which can be hard to follow as they give little context to the feature that the commit is being added for.
+
+## Good Pull Request titles
+
+When creating a pull request, it's important to have a clear and descriptive title. This is important when looking back at the git history, as it allows you to see at a glance what changes were made in a particular pull request.
```
Comment (danielmackay) [inline rules/clean-git-history/rule.md:38]: Should we add something about gitmoji?  I find helps to quickly scan a branch and see what sort of changes are going in.  

I think we have a rule about PBI titles.  Do we have one about PR titles?  Perhaps the PBI titles rule applies here too?

**Review by danielmackay — APPROVED** [review]
Approved.  One comment.

File: rules/clean-git-history/rule.md (hunk 1 above)
Comment (GordonBeeming) [reply rules/clean-git-history/rule.md:38]: We have a rule on PR titles, I've linked it in the related rules... I can move it inline here if that would be better

Comment (GordonBeeming) [reply rules/clean-git-history/rule.md:38]: As per our conversation, the writing good PRs rule in the related rules covers what's good for titles so we don't need to repeat it here
